app/models/course.rb

```diff
@@ -377,7 +377,8 @@ def sites=(desired_sites)
     to_remove = existing_sites - desired_sites
     to_remove.each { |site| remove_site!(site: site) }
 
-    sites.reload
+    sites.reload if persisted?
```

Comment:
How about an `if ... else` here to make it clearer?

Reply:
Yeah I was thinking that also - i'll change that and merge